app/com/gu/contentapi/sanity/package.scala

```diff
@@ -63,23 +64,29 @@ package object sanity extends ScalaFutures with Matchers with IntegrationPatienc
         else {
           incidentCount match {
             case `pagerDutyThreshold` =>
-              println("reporting")
-              pagerDutyAlerter(testName, tfe, tags, getIncidentKey)
+              //reset incident count in next incident after threshold has been met
               resetCount
             case _ =>
-              // increment the incident count only if there is an existing recent incident {
+              // increment the incident count only if there is an existing recent incident or it is the first incident {
               if (lastIncidentDateTime.isEmpty || (Minutes.minutesBetween(lastIncidentDateTime.get, DateTime.now).getMinutes < 10)) {
+                //increment counter
                 incrementIncidentCount
-                lastIncidentDateTime = Some(DateTime.now)
+                if (incidentCount == pagerDutyThreshold)
+                //report when threshold is met
+                {
+                  println("Reporting")
+                  pagerDutyAlerter(testName, tfe, tags, getIncidentKey)
+                }
               }
               else {
+                //reset counter if incident is not recent
                 resetCount
               }
-            }
           }
-          fail
         }
+        fail
```

Comment:
Maybe I'm misreading this, but this code seems to be overcomplicating things.

Instead of

```
          incidentCount match {
            case `pagerDutyThreshold` =>
              //reset incident count in next incident after threshold has been met
              resetCount
            case _ =>
              // increment the incident count only if there is an existing recent incident or it is the first incident {
              if (lastIncidentDateTime.isEmpty || (Minutes.minutesBetween(lastIncidentDateTime.get, DateTime.now).getMinutes < 10)) {
                //increment counter
                incrementIncidentCount
                if (incidentCount == pagerDutyThreshold)
                //report when threshold is met
                {
                  println("Reporting")
                  pagerDutyAlerter(testName, tfe, tags, getIncidentKey)
                }
              }
              else {
                //reset counter if incident is not recent
                resetCount
              }
          }
```

can't you reset the counter after you send a pager duty alert? e.g.

```
              // increment the incident count only if there is an existing recent incident or it is the first incident {
              if (lastIncidentDateTime.isEmpty || (Minutes.minutesBetween(lastIncidentDateTime.get, DateTime.now).getMinutes < 10)) {
                //increment counter
                incrementIncidentCount
                if (incidentCount == pagerDutyThreshold)
                //report when threshold is met
                {
                  println("Reporting")
                  pagerDutyAlerter(testName, tfe, tags, getIncidentKey)
                  resetCount // NOTE: reset the counter after sending the alert
                }
              }
              else {
                //reset counter if incident is not recent
                resetCount
              }
```